extension/settingsproviderextension/factory.go

```diff
@@ -0,0 +1,56 @@
+package settingsproviderextension
+
+import (
+	"context"
+	"fmt"
+	stsSettingsConfig "github.com/stackvista/sts-opentelemetry-collector/extension/settingsproviderextension/internal"
+	"github.com/stackvista/sts-opentelemetry-collector/extension/settingsproviderextension/provider/file"
+	stsSettingsSource "github.com/stackvista/sts-opentelemetry-collector/extension/settingsproviderextension/provider/kafka"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/extension"
+)
+
+var (
+	Type = component.MustNewType("settings_provider")
+)
+
+func NewFactory() extension.Factory {
+	return extension.NewFactory(
+		Type,
+		createDefaultConfig,
+		createExtension,
+		// TODO: eventually this needs to become StabilityLevelStable
+		component.StabilityLevelDevelopment,
+	)
+}
+
+func createDefaultConfig() component.Config {
+	return &stsSettingsConfig.Config{
+		File: &stsSettingsConfig.FileSettingsProviderConfig{
+			Path: "./testdata/otel_mappings.yaml",
```

Comment:
I would not keep this as the default config (maybe that's not the intention), better to fail startup if there is no config than to use test data.